### PR TITLE
Development

### PR DESCRIPTION
**Pull request by dpak1999**
@sagarchoudhary96 review 🙂🙂🙂

### CODE REVIEW

File: src/Components/ModalComponent.js [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+  };
+
+  useEffect(() => {
+    const data = JSON.parse(localStorage.getItem("note"));
+    setNotes(data);
+  }, []);
+
+  return (
+    <div>
+      <Modal
```
Comment (sagarchoudhary96) [inline src/Components/ModalComponent.js:23]: Please refactor this as discussed,  Modal is something that opens up as a dialog In your case on click of add task you are opening modal which should just have the form and the list should ideally be outside the modal in your home component.

File: src/Components/NotesList.js [hunk 2]
```diff
@@ -0,0 +1,25 @@
+import React from "react";
+import { Card, Col, Container, Row } from "react-bootstrap";
+
+const NotesList = ({ notes }) => {
+  const note = notes.map((item) => {
+    return (
+      <Col md={6} key={Math.random()}>
```
Comment (sagarchoudhary96) [inline src/Components/NotesList.js:7]: don't use `Math.random()` as key instead while creating a note, add a 'note_id' by yourself to note object. which can be a unique id for each note. you can use https://www.npmjs.com/package/uuid package

**Review by sagarchoudhary96 — CHANGES_REQUESTED** [review]

**Comment by dpak1999** [comment]
@sagarchoudhary96  review please

File: src/Components/HomeComponent.js [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+    const data = JSON.parse(localStorage.getItem("note"));
+    if (data) {
+      setNotes(data);
+
+      const result = data.filter(({ title }) =>
+        title.toLowerCase().includes(searchItem.toLowerCase())
+      );
+      setResults(result);
+    }
+  }, [searchItem]);
```
Comment (sagarchoudhary96) [inline src/Components/HomeComponent.js:33]: you need to split this hook into too. 

1. componentDidMount lifecycle --> it should have the part to get data from localStorage and update result and notes arr with same data.

2. useEffect with dependency on searchItem -> this should have the logic to filter notes based on search item.

File: src/Components/HomeComponent.js [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+      </div>
+      <ModalComponent
+        show={modalShow}
+        onHide={() => setModalShow(false)}
+        addNote={addNote}
+      />
+      {searchItem ? (
+        <SearchResults results={results} />
+      ) : (
+        <NotesList notes={notes} />
```
Comment (sagarchoudhary96) [inline src/Components/HomeComponent.js:60]: same notesList component can be used here right, not need to create searchResults component. you can just pass `notes={results}` to your component.

File: src/Components/SearchResults.js [hunk 5]
```diff
@@ -0,0 +1,26 @@
+/** @format */
+
+import React from "react";
```
Comment (sagarchoudhary96) [inline src/Components/SearchResults.js:3]: this whole component can be removed.

**Review by sagarchoudhary96 — CHANGES_REQUESTED** [review]

**Comment by dpak1999** [comment]
@sagarchoudhary96  done refactoring

**Review by sagarchoudhary96 — COMMENTED** [review]
Looks good now

**Pull request merged**